### DISCUSSION TITLE
Column chart: Difficulties displaying Line Components with rounded corners

### DISCUSSION THREAD

**Discussion by Itzekey** [patrykandpatrick/vico#813] [Questions] [answered]
### Question

Hello, is it possible to display column charts where each column doesn't necessarily use the same number of line components as other columns, while having rounded corners for each line component located both at the bottom and the top?

I created a function that generates line components based on the data list provided. It calculates the shape for each line component based on the index of the items from the data list.

My logs print out the correct shapes for each line component, but visually it doesn't add up.

I wanted to know if I'm mistaken (which is most probably the case) or if it's not possible to have different numbers of line components stacked on each other for each column.

![Screenshot 2024-07-24 204801](https://github.com/user-attachments/assets/675fd5ea-82d4-418e-94dc-fb0ab805d03e)

This is how the Sample Chart looks.

![Screenshot 2024-07-24 204957](https://github.com/user-attachments/assets/8684b5fb-11da-446d-9bde-c88b5ca4ef54)

This is how my chart looks, and my question is whether the line components at the top not being round is due to not using the same number of line components for each x-value.

### Vico version(s)

2.0.0-alpha.14

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello @Itzekey. As mentioned in the documentation, `ColumnProvider.series` associates `LineComponent`s with entire series, not individual columns. The first `LineComponent` is used for each entry of the first series, the second `LineComponent` is used for each entry of the second series, and so on. It is possible to achieve the result you're looking for, but you have to create a suitable `ColumnProvider`. `ColumnProvider.series` would be appropriate for achieving this effect if the first and last series had no gaps.
